tests/integration/test_client.py

```diff
@@ -57,3 +61,49 @@ async def test_client_start_transcribe_stream_bad_request(self, client):
                 media_sample_rate_hz=9999999,
                 media_encoding="pcm",
             )
+
+    @pytest.mark.asyncio
+    async def test_client_start_transcribe_stream_incompatible_args(self, client):
+        # show_speaker_label and enable_channel_identification should
+        # not be both set when making StartStreamTranscriptionRequest.
+        # Test that exception raised by service.
+        with pytest.raises(BadRequestException):
+            stream = await client.start_stream_transcription(
+                language_code="en-US",
+                media_sample_rate_hz=16000,
+                media_encoding="pcm",
+                show_speaker_label="true",
+                enable_channel_identification="true",
+                number_of_channels=2,
+            )
+
+    @pytest.mark.asyncio
+    async def test_start_transcribe_stream_bad_boolean_show_speaker_lab(self, client):
+        with pytest.raises(SerializationException):
+            stream = await client.start_stream_transcription(
+                language_code="en-US",
+                media_sample_rate_hz=16000,
+                media_encoding="pcm",
+                show_speaker_label="foo",
+                enable_channel_identification="true",
+            )
+        with pytest.raises(SerializationException):
+            stream = await client.start_stream_transcription(
+                language_code="en-US",
+                media_sample_rate_hz=16000,
+                media_encoding="pcm",
+                show_speaker_label="true",
+                enable_channel_identification="foo",
+                number_of_channels=2,
+            )
+
+    @pytest.mark.asyncio
+    async def test_start_transcribe_stream_boolean_miscasing(self, client):
+        with pytest.raises(BadRequestException):
+            stream = await client.start_stream_transcription(
+                language_code="en-US",
+                media_sample_rate_hz=16000,
+                media_encoding="pcm",
+                show_speaker_label="TrUe",
```

Comment:
Miscasing the values raises an exception? We should probably try to smooth that out during serialization. How do we handle booleans being provided for these values too?

Reply:
The thought process of this was to have two mis-cased trues sent in and the service does sanitation on the booleans. Then it sends back the BadRequestException for having a true set for both show_speaker_label and enable_channel_identification. Therefore validating the casing was indeed fixed on the service end and expected exception resulted.

Reply:
Ah ok, so that isn't clear from the test name since it seems to be about casing. Since we already had that test above for testing both values being true, we wouldn't need to do it again here. It's best to keep tests as discrete as possible so it's clear what's causing the failure.